handler/core/src/main/java/io/knotx/fragments/task/factory/node/action/ActionNodeFactory.java

```diff
@@ -46,21 +54,30 @@ public String getName() {
 
   @Override
   public ActionNodeFactory configure(JsonObject config, Vertx vertx) {
-    actionProvider = new ActionProvider(supplyFactories(),
-        new ActionNodeFactoryConfig(config).getActions(), vertx);
+    this.actionNameToOptions = new ActionNodeFactoryConfig(config).getActions();
+    actionProvider = new ActionProvider(supplyFactories(), actionNameToOptions, vertx);
     return this;
   }
 
   @Override
   public Node initNode(GraphNodeOptions nodeOptions, Map<String, Node> edges,
       NodeProvider nodeProvider) {
+    // The implementation is for backwards compatibility of NodeFactory interface
+    return initNode(nodeOptions, edges, nodeProvider, new HashMap<>());
+  }
```

Comment:
Maybe it should just have a default implementation in the interface. I think backward compatibility is why default implementations were invented